AVsitter2/Plugins/AVcontrol/LockMeister/[AV]LockMeister.lsl

```diff
@@ -0,0 +1,220 @@
+        PSYS_PART_START_SCALE, < 0.05, 0.05, 0.0 >,
+        PSYS_PART_MAX_AGE, 3.0,
+        PSYS_SRC_BURST_RATE, 0.001,
+        PSYS_SRC_TEXTURE, "6e389f5c-e34d-53f0-e607-eba6fcfd0e58",
+        PSYS_SRC_BURST_SPEED_MIN,0.1,
+        PSYS_SRC_BURST_SPEED_MAX,0.1,
+        PSYS_SRC_TARGET_KEY, target,
+        PSYS_SRC_BURST_PART_COUNT, 1,
+        PSYS_SRC_ACCEL, <0,0,-0.3>
+        ]);
+}
+
+list    emitter_links;
+list    emitter_states;
+list    emitter_lmcodes;
+integer emitter_v2_count;
+
+ClearChains()
+{
+    llSetTimerEvent(0.0);
+    llListenControl(lm_handle,FALSE);
+
+    integer chain = llGetListLength(emitter_links);
+    while(chain--)
+        llLinkParticleSystem(llList2Integer(emitter_links,chain),[]);
+
+    avatar = NULL_KEY;
+    emitter_links=[];
+    emitter_states=emitter_links;
+    emitter_lmcodes=emitter_links;
+    emitter_v2_count=0;
+}
+
+integer lm_channel = -8888;
+integer lm_handle;
+key avatar;
+
+list ring_prims = [avatar]; //OSS::list ring_prims; // Force error if not compiled in Mono
+
+integer TIMEOUT = 10;       //How long do we wait for a reply before closing communications.
+
+default
+{
+	state_entry()
+	{
+		lm_handle = llListen(lm_channel,"","","");
+        llListenControl(lm_handle,FALSE);
+	}
+    link_message(integer sender, integer num, string msg, key id)
+    {
+        if (sender != llGetLinkNumber()) return;
+
+        if (num == 90065)
+        {
+            //If OUR user stood up, we clear all drawn chains.
+
+            if (id != avatar) return;
+            ClearChains();
+        }
+        else if (num == 90045)
+        {
+            //If a new animation is being played, we have to clear existing chains (if needed),
+            //and draw new ones if the animation is registered in POSES.
+
+            list payload = llParseString2List(msg, ["|"], []);
+
+            if(SITTER != (integer)llList2String(payload, 0)) return; //payload:0 == which seat number this is for.
+
+            if (avatar)
+                ClearChains();
+
+            avatar = id;
+
+            string POSE_NAME = llList2String(payload, 1);
+            integer pose_index = llListFindList(POSES, [POSE_NAME]);
+
+            if(!~pose_index) //If we haven't found a named pose, try to find the wildcard (*) pose.
+            {
+                pose_index = llListFindList(POSES, ["*"]);
+                if(!~pose_index) return;
+            }
+
+            llListenControl(lm_handle,TRUE);
+            llSetTimerEvent(TIMEOUT);
+
+            list data = llCSV2List(llList2String(POSES,pose_index+1));
+            integer i;
+            for(;i<llGetListLength(data);i+=2)
+            {
+                string lm_target = llList2String(data,i);
+                string prim_name = llList2String(data,i+1);
+
+                //lookup the prim linknumber of this anchor point.
+                integer j = llGetNumberOfPrims();
+                while(j--)
+                {
+                    integer link_id = j+1;
```

Comment:
This does not work for 1 prim objects. Some clever mesher could try to use it with 1 prim. Please make it work with 1 prim.
```suggestion
                integer j = llGetNumberOfPrims();
                integer ofs = j > 1;
                while (j--)
                {
                    integer link_id = j+ofs;
```